connect_four/lib/board.rb

```diff
@@ -0,0 +1,138 @@
+require 'colorize'
+require_relative 'constants'
+
+class Board
+	attr_accessor :number_of_pos_filled, :board_array
+	attr_reader :number_of_rows, :number_of_cols
+
+	def initialize(num_rows = 6, num_cols = 7)
+		@number_of_pos_filled = 0
+		@number_of_rows = num_rows
+		@number_of_cols = num_cols
+		@board_array = []
+		@board_array[0] = [0,0,0,0,0,0,0]
+		@board_array[1] = [0,0,0,0,0,0,0]
+		@board_array[2] = [0,0,0,0,0,0,0]
+		@board_array[3] = [0,0,0,0,0,0,0]
+		@board_array[4] = [0,0,0,0,0,0,0]
+		@board_array[5] = [0,0,0,0,0,0,0]
+	end
+
+	def board_array_column
+		@board_array.transpose
+	end
+
+	def available_columns
+		@board_array[0].each_index.select{|i| @board_array[0][i] == NO_COLOR}
+	end
+
+	def remove_piece(col)
+		index_to_remove = board_array_column[col].each_index.select{|i| board_array_column[col][i] == NO_COLOR}.max
+		if index_to_remove
+			@board_array[index_to_remove + 1][col] = NO_COLOR
+		else
+			@board_array[0][col] = NO_COLOR
+		end
+	end
+
+	def add_piece(color, col)
+		if board_array_column[col][0] == NO_COLOR
+			index_to_add = board_array_column[col].each_index.select{|i| board_array_column[col][i] == NO_COLOR}.max
+			@board_array[index_to_add][col] = color
+			@number_of_pos_filled += 1
+			true
+		else
+			false
+		end
+	end
+
+	def check_line(line, color)
+		return_val = false
+		possible_win_pos = line.size - 3
+		possible_win_pos.times do |index|
+			start_pos = index
+			end_pos = index + 3
+			if line[start_pos..end_pos].all? {|item| item == color}
```

Comment:
this `if` block can be replaced by `return true if line[start_pos..end_pos].all? {|item| item == color}`